pkg/storage/stores/shipper/compactor/table.go

```diff
@@ -61,7 +61,7 @@ func newTable(ctx context.Context, workingDirectory string, objectClient chunk.O
 }
 
 func (t *table) compact() error {
-	objects, _, err := t.storageClient.List(t.ctx, t.name+"/")
```

Comment:
The forward slash here needs to stay because we are trying to list contents of a directory without it we will get the name of the same directory back with hosted object stores. This is due to the object stores not having a concept of directories.